test/gossip-incoming.js

```diff
@@ -0,0 +1,140 @@
+/* eslint-env mocha */
+/* eslint max-nested-callbacks: ["error", 5] */
+'use strict'
+
+const chai = require('chai')
+chai.use(require('dirty-chai'))
+chai.use(require('chai-spies'))
+const expect = chai.expect
+
+const {
+  createNode,
+  dialNode,
+  startNode,
+  stopNode
+} = require('./utils')
+
+const shouldNotHappen = (msg) => expect.fail()
+
+describe('gossip incoming', () => {
+  const topic = 'Z'
+  let nodeA
+  let nodeB
+  let nodeC
+
+  describe('gossipIncoming == true', () => {
+    beforeEach(async function () {
+      this.timeout(4000)
+      nodeA = await createNode('/ip4/127.0.0.1/tcp/0')
+      nodeB = await createNode('/ip4/127.0.0.1/tcp/0')
+      nodeC = await createNode('/ip4/127.0.0.1/tcp/0')
```

Comment:
As a suggestion, I have been starting to use the following for this cases:

```js
[nodeA, nodeB, nodeC] = await Promise.all([
  createNode('/ip4/127.0.0.1/tcp/0'),
  createNode('/ip4/127.0.0.1/tcp/0'),
  createNode('/ip4/127.0.0.1/tcp/0')
])
```

I think it is more readable because we are grouping lines with the same responsibility. Also, this works in parallel which will be faster (as this is a test, it is no more than a suggestion).

I would apply the `Promise.all()` also to the `startNode` below and the `dialNode`

Reply:
Thanks for the suggestion :) 
Definitely seems better with the Promise.all.
I'm going to hold off for this PR, as this is somewhat urgent (and we're wanting to cut a new release for our eth2 interop starting tomorrow).
But duely noted!

Reply:
go for it 👍